presto-main/src/test/java/io/prestosql/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java

```diff
@@ -0,0 +1,407 @@
+        acceptButton.click();
+    }
+
+    private void assertPrestoCookie(Cookie cookie)
+    {
+        assertThat(cookie.getName()).isEqualTo("Presto-OAuth2-Token");
+        assertThat(cookie.getDomain()).isEqualTo("host.testcontainers.internal");
+        assertThat(cookie.getPath()).isEqualTo("/ui/");
+        assertThat(cookie.isSecure()).isTrue();
+        assertThat(cookie.isHttpOnly()).isTrue();
+        assertThat(cookie.getValue()).isNotBlank();
+        Jws<Claims> jwt = Jwts.parser()
+                .setSigningKeyResolver(new JwkSigningKeyResolver(new JwkService(
+                        new JwtAuthenticatorConfig().setKeyFile("http://localhost:" + testingHydraService.getHydraPort() + "/.well-known/jwks.json"),
+                        new JettyHttpClient())))
+                .parseClaimsJws(cookie.getValue());
+        // TODO: check source of the difference
+        //        assertThat(cookie.getExpiry()).isEqualTo(jwt.getBody().getExpiration());
+        assertAccessToken(jwt);
+    }
+
+    private void assertAccessToken(Jws<Claims> jwt)
+    {
+        assertThat(jwt.getBody().getSubject()).isEqualTo("foo@bar.com");
+        assertThat(jwt.getBody().get("client_id")).isEqualTo("another-consumer");
+        assertThat(jwt.getBody().getIssuer()).isEqualTo("http://hydra:4444/");
+    }
+
+    private void assertUICallWithCookie(Cookie cookie)
+            throws IOException
+    {
+        OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder();
+        setupInsecureSsl(httpClientBuilder);
+        httpClientBuilder.followRedirects(false);
+        httpClientBuilder.cookieJar(new CookieJar()
+        {
+            @Override
+            public void saveFromResponse(HttpUrl url, List<okhttp3.Cookie> cookies)
+            {
+            }
+
+            @Override
+            public List<okhttp3.Cookie> loadForRequest(HttpUrl url)
+            {
+                return ImmutableList.of(
+                        new okhttp3.Cookie.Builder()
+                                .domain("localhost")
+                                .path("/ui/")
+                                .name("Presto-OAuth2-Token")
+                                .value(cookie.getValue())
+                                .httpOnly()
+                                .secure()
+                                .build());
+            }
+        });
+
+        // pass access token in Presto UI cookie
+        assertThat(httpClientBuilder.build().newCall(uiCall().build())
+                .execute()
+                .code())
+                .isEqualTo(OK.getStatusCode());
+    }
+
+    private static int findAvailablePort()
+    {
+        try (ServerSocket tempSocket = new ServerSocket(0)) {
+            return tempSocket.getLocalPort();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static void waitForNodeRefresh(TestingPrestoServer server)
```

Comment:
Let's make this a method on `TestingPrestoServer` since we do this in several places now